src/translations/maps/indexMap.js

```diff
@@ -0,0 +1,150 @@
+import { rangeEach } from '../../helpers/number';
+import { mixin } from '../../helpers/object';
+import { isFunction } from '../../helpers/function';
+import localHooks from '../../mixins/localHooks';
+
+/**
+ * Map for storing mappings from an index to a value.
+ */
+class IndexMap {
```

Comment:
As for `IndexMapper`, `IndexMap` doesn't have TS definition. Maybe this class is hidden so deeply that no TS definition is necessary for a developer who writes in TS?